R/bootstrap_traits_parametric.R

```diff
@@ -0,0 +1,94 @@
+#' Bootstrap traits parametrically
+#' @description Bootstrap fitted distributions
+#' @param fitted_distributions
+#' Fitted distribution object returned by fit_trait_distributions
+#' @param nrep number of bootstrap replicates
+#' @param sample_size number of plants per sample
+#' @description
+#'
+#' @return a tibble
+#'
+#' @importFrom stats var
+#' @importFrom e1071 skewness kurtosis
+#' @importFrom magrittr %>%
+#' @importFrom dplyr slice_sample group_by summarise rowwise
+#' @importFrom tidyr unnest
+#' @importFrom purrr map_dfr
+#' @examples
+#' data(community)
+#' data(trait)
+#' imputed_traits <-trait_impute(comm = community, traits = trait,
+#'                  scale_hierarchy = c("Site", "PlotID"),
+#'                  taxon_col = "Taxon", value_col = "Value",
+#'                  trait_col = "Trait", abundance_col = "Cover")
+#' fitted_distributions <- fit_trait_distributions(imputed_traits =
+#' imputed_traits, distribution_type = "normal")
+#' parametric_dists <- trait_parametric_bootstrap(fitted_distributions =
+#' fitted_distributions, nrep = 10, sample_size = 100)
+#' moment_summary <-
+#' trait_summarise_boot_moments(bootstrap_moments = parametric_dists,
+#' parametric = FALSE)
+#' @export
+trait_parametric_bootstrap <- function(fitted_distributions,
+                                       nrep = 100,
+                                       sample_size = 200) {
+
+  #Check that inputs makes sense
+  if (!"parametric_distributions" %in% class(fitted_distributions)) {
+    stop("Fitted distributions not properly formatted.
+         Please use fit_trait_distributions()")
+  }
+
+  if (!all(is.numeric(nrep) & is.numeric(sample_size))) {
```

Comment:
more strictly, is.integer, "should be integers"

Reply:
Unfortunately is.integer() will fail on anything that is entered with the L..  For example, `is.integer(4L) == TRUE` works, but not `is.integer(4)  == F`.  We could use `nrep %% 1 == 0` or `nrep == as.integer(nrep)`? For now I've taken the former approach using %% to check the fractional part of nrep, but let me know if you have a preferred solution.